src/Ad.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Created by kenshoo-linux on 14/07/16.
+ */
+
+/**
+ * Created by ken-linux2 on 12/07/16.
+ */
+public class Ad {
+
+    private String engine;
```

Comment:
Rename to AdGroup.
The business key here will be engine, type, theme.
In the DB exercise part you will need for fetch and connect to this entity if already existing (start with insert).